pkg/controller/account/account_controller.go

```diff
@@ -768,3 +851,73 @@ func (r *ReconcileAccount) statusUpdate(reqLogger logr.Logger, account *awsv1alp
 	reqLogger.Info(fmt.Sprintf("Status updated for %s", account.Name))
 	return err
 }
+
+func createCase(reqLogger logr.Logger, accountID string, client awsclient.Client) (string, error) {
+	// Initialize basic communication body and case subject
+	caseCommunicationBody := fmt.Sprintf("Hi AWS, please add this account to Enterprise Support: %s\n\nAlso please apply the following limit increases:\n\n", accountID)
+	caseSubject := fmt.Sprintf("Add account %s to Enterprise Support and increase limits", accountID)
+
+	// For each supported AWS region append to the communication a request of limit increase
+	for index, region := range coveredRegions {
+		caseLimitIncreaseBody := fmt.Sprintf("Limit increase request %d\nService: EC2 Instances\nRegion: %s\nPrimary Instance Type: %s\nLimit name: Instance Limit\nNew limit value: %d\n------------\n", index+1, region, desiredInstanceType, caseDesiredInstanceLimit)
+		caseCommunicationBody += caseLimitIncreaseBody
+	}
+
+	createCaseInput := support.CreateCaseInput{
+		CategoryCode:      aws.String(caseCategoryCode),
+		ServiceCode:       aws.String(caseServiceCode),
+		IssueType:         aws.String(caseIssueType),
+		CommunicationBody: aws.String(caseCommunicationBody),
+		Subject:           aws.String(caseSubject),
+		SeverityCode:      aws.String(caseSeverity),
+	}
+
+	reqLogger.Info("Creating the case", "CaseInput", createCaseInput)
+
+	caseResult, caseErr := client.CreateCase(&createCaseInput)
```

Comment:
We don't use it everywhere yet but I'm slowing trying to convert all the error checks to using AWS SDK error interface and examples here: https://docs.aws.amazon.com/sdk-for-go/api/aws/awserr/

Reply:
Done